HKMP/Game/Client/PlayerManager.cs

```diff
@@ -37,9 +43,39 @@ internal class PlayerManager {
         public Team LocalPlayerTeam { get; private set; } = Team.None;
 
         /// <summary>
-        /// The player prefab GameObject.
+        /// The player container prefab GameObject.
+        /// </summary>
+        private GameObject _playerContainerPrefab;
+
+        /// <summary>
+        /// A collection of pre-instantiated players that will be fetched when spawning a player.
+        /// </summary>
+        private readonly ConcurrentBag<GameObject> _inactivePlayers = new();
+
+        /// <summary>
+        /// The collection of active players spawned from and not in the player pool.
         /// </summary>
-        private readonly GameObject _playerPrefab;
+        private readonly Dictionary<ushort, GameObject> _activePlayers = new();
+
+        /// <summary>
+        /// The initial size of the pool of player container objects to be pre-instantiated.
+        /// </summary>
+        private const ushort InitialPoolSize = 256;
+
+        private readonly List<Type> _usernameComponentTypes = new();
```

Comment:
Please initialize non-static members in the constructor rather than at they definition.
Also, this member is missing documetation.